odoo/tools/misc.py

```diff
@@ -1790,3 +1790,24 @@ def get_flag(country_code: str) -> str:
 def format_frame(frame):
     code = frame.f_code
     return f'{code.co_name} {code.co_filename}:{frame.f_lineno}'
+
+
+def named_to_positional_printf(string: str, args: Mapping) -> tuple[str, tuple]:
+    """ Convert a named printf-style format string with its arguments to an
+    equivalent positional format string with its arguments.
+    """
+    args = _PrintfArgs(args)
+    return string % args, tuple(args.values)
+
+
+class _PrintfArgs:
+    """ Helper object to turn a named printf-style format string into a positional one. """
+    __slots__ = ('mapping', 'values')
+
+    def __init__(self, mapping):
+        self.mapping = mapping
+        self.values = []
+
+    def __getitem__(self, key):
+        self.values.append(self.mapping[key])
+        return "%s"
```

Comment:
I think we need to do something for the wildcard characters 🤔 
`named_to_positional_printf('%(percent)s%%s', {'percent': '100'}) == ('%s%s', ('100',))`

Reply:
They are simply not supported.  I will document that.